### PR TITLE
[FSK-22]: Initialize NestJS Repo

### PR DESCRIPTION
**Pull request by MohammadForoutan**
*   configure tsconfig.json
*   add configModule
*   configure environment variables
*   add test for the app module

### CODE REVIEW

**Review by mhmda-83 — APPROVED** [review]

File: .example.env [hunk 1]
```diff
@@ -0,0 +1,2 @@
+PORT=
+HOST=
```
Comment (ASafaeirad) [inline .example.env]: Let's keep a new line at the end of all files, as mentioned in the POSIX standard.

File: .gitignore [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+*.launch
+.settings/
+*.sublime-workspace
+
+# IDE - VSCode
+.vscode/*
+!.vscode/settings.json
+!.vscode/tasks.json
+!.vscode/launch.json
+!.vscode/extensions.json
```
Comment (ASafaeirad) [inline .gitignore]: Let's keep a new line at the end of all files, as mentioned in the POSIX standard.

File: .npmrc [hunk 3]
```diff
@@ -0,0 +1,2 @@
+save-exact=true
+update-notifier=false
```
Comment (ASafaeirad) [inline .npmrc]: Let's keep a new line at the end of all files, as mentioned in the POSIX standard.

File: .prettierrc [hunk 4]
```diff
@@ -0,0 +1,4 @@
+{
+  "singleQuote": true,
+  "trailingComma": "all"
+}
```
Comment (ASafaeirad) [inline .prettierrc]: Let's keep a new line at the end of all files, as mentioned in the POSIX standard.

File: package.json [hunk 5]
```diff
@@ -0,0 +1,75 @@
...
+    "testRegex": ".*\\.spec\\.ts$",
+    "transform": {
+      ".+\\.(t|j)s$": "ts-jest"
+    },
+    "collectCoverageFrom": [
+      "**/*.(t|j)s"
+    ],
+    "coverageDirectory": "../coverage",
+    "testEnvironment": "node"
+  }
```
Comment (ASafaeirad) [inline package.json]: It's a good practice to keep jest config in a separate folder, as It's possible to cache the install step. Changing package.json invalidates that cache.

File: src/config/configuration.ts [hunk 6]
```diff
@@ -0,0 +1,29 @@
+import { registerAs } from '@nestjs/config';
+import { IAppConfig } from '../../@types/env';
```
Comment (ASafaeirad) [inline src/config/configuration.ts]: It's not a good idea to import something outside the `src` boundary.
It's better to keep relevant information as close as possible and move `IAppConfig` into the config folder.

**Review by ASafaeirad — CHANGES_REQUESTED** [review]

**Review by ASafaeirad — APPROVED** [review]

**Pull request merged**